### PR TITLE
fix error message disappear

### PR DESCRIPTION
**Pull request by dukenv0307**
### Details 

- This PR will fix the issue: Scan - Merchant message disappears after deleting receipt and reappears after adding receipt
### Fixed Issues

$ https://github.com/Expensify/App/issues/34997
PROPOSAL: https://github.com/Expensify/App/issues/34997#issuecomment-1907046052

### Tests

1. Go to workspace chat
2. Create a scan request
3. While the scanning is ongoing, navigate to the request details page and manually enter amount
4. Click on the receipt and delete it
5. Verify that the error and RBR stay at the 'Merchant' field
6. Add a receipt
7. Verify that the error and RBR stay at the 'Merchant' field
- [x] Verify that no errors appear in the JS console

### QA Steps

1. Go to workspace chat
2. Create a scan request
3. While the scanning is ongoing, navigate to the request details page and manually enter amount
4. Click on the receipt and delete it
5. Verify that the error and RBR stay at the 'Merchant' field
6. Add a receipt
7. Verify that the error and RBR stay at the 'Merchant' field
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

[Screencast from 07-02-2024 18:05:28.webm](https://github.com/Expensify/App/assets/129500732/fb630286-1ba1-4cd6-b563-7a4ef64ee679)

</details>

<details>
<summary>Android: mWeb Chrome</summary>

[Screencast from 06-02-2024 11:59:58.webm](https://github.com/Expensify/App/assets/129500732/0fb9b2f9-df70-4a4c-9243-0e824e4db6b5)

</details>

<details>
<summary>iOS: Native</summary>

[Screencast from 06-02-2024 13:29:55.webm](https://github.com/Expensify/App/assets/129500732/c22d8a4f-3e46-4258-bef3-ec845d150fcd)

</details>

<details>
<summary>iOS: mWeb Safari</summary>

[Screencast from 06-02-2024 11:57:55.webm](https://github.com/Expensify/App/assets/129500732/b89131f3-810f-40bc-9615-f6b0063909de)

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

[Screencast from 06-02-2024 11:45:05.webm](https://github.com/Expensify/App/assets/129500732/df9b4116-8d60-4a2a-93ce-aae55c86546d)

</details>

<details>
<summary>MacOS: Desktop</summary>

[Screencast from 06-02-2024 13:24:32.webm](https://github.com/Expensify/App/assets/129500732/88c5e1d5-4a3b-45c4-9a68-33c644b76b80)

</details>

### CODE REVIEW

**Comment by c3024** [comment]
Please merge main. Typechecks are fixed on it.

**Comment by c3024** [comment]
I think RBR on LHN as well should be fixed within this issue.

https://github.com/Expensify/App/assets/102477862/ee819e26-7ddc-4e33-94a7-fb236332dd88

@dukenv0307 @Julesssss

**Comment by Julesssss** [comment]
> I think RBR on LHN as well should be fixed within this issue.

Ah, so that occurs with this PR, but not on main? If so I agree.

**Comment by c3024** [comment]
No this PR did not introduce it. The behaviour is already there are on `main`.

**Comment by c3024** [comment]
> No, this PR did not introduce it. The behaviour is already there are on `main`.

When there is an error shown at a mandatory field, I think it is logical to show RBR on LHN as well. That's why I asked if showing RBR on LHN should be included within the scope of this issue. This PR did not change any behaviour that already existed on `main` w.r.t. LHN.

@Julesssss

**Comment by Julesssss** [comment]
> When there is an error shown at a mandatory field, I think it is logical to show RBR on LHN as well. That's why I asked if showing RBR on LHN should be included within the scope of this issue

Okay got it.

@dukenv0307 could you please apply a change to your PR to ensure the error is shown on the LHN?

**Comment by c3024** [comment]
I think once a request is deemed as a scan request, even if we remove the receipt, the error at the merchant field and RBRs on LHN should be same as when there was a receipt. 

https://github.com/Expensify/App/assets/102477862/fd51f3f8-5650-46de-9a89-13139ba2bc35

Here, on member and admin side together there are total three red dots and zero green dots when there is a receipt. When the receipt is removed, IMO there should be the same dots as earlier. But there is one red dot and one green dot. @dukenv0307 

cc: @Julesssss

**Comment by Christinadobrzyn** [comment]
Hi there, can you check to see if this [issue](https://github.com/Expensify/App/issues/36574) might be a regression from this PR?

**Comment by c3024** [comment]
Those changes work well. Please include them in the PR. @dukenv0307

**Comment by c3024** [comment]
Please merge `main` as well because we need these changes included in this PR #36810 also to test correctly. @dukenv0307

**Comment by Julesssss** [comment]
Seems like we're nearly there 👍

File: src/libs/TransactionUtils.ts [hunk 1]
```diff
@@ -447,7 +447,14 @@ function isReceiptBeingScanned(transaction: OnyxEntry<Transaction>): boolean {
...
  */
 function hasMissingSmartscanFields(transaction: OnyxEntry<Transaction>): boolean {
-    return Boolean(transaction && hasReceipt(transaction) && !isDistanceRequest(transaction) && !isReceiptBeingScanned(transaction) && areRequiredFieldsEmpty(transaction));
+    return Boolean(transaction && !isDistanceRequest(transaction) && !isReceiptBeingScanned(transaction) && areRequiredFieldsEmpty(transaction));
+}
+
+/**
+ * Check if the transaction has missing required fields
+ */
+function hasMissingRequiredFields(transaction: OnyxEntry<Transaction>): boolean {
```
Comment (c3024) [inline src/libs/TransactionUtils.ts:456]: I think it is better to just use `hasMissingSmartScanFields` at the places where we added `hasMissingRequiredFields` even though the name `hasMissingSmartScanFields` might be a little misleading for other requests like manual requests with receipts. 

Adding another function with a different name like this `hasMissingRequiredFields` that returns the same value can be more confusing later IMO. @dukenv0307 

cc: @Julesssss

Comment (Julesssss) [reply src/libs/TransactionUtils.ts:456]: I'm on the fence. Having a better-described function name is good. But we can always do that later, or rename the existing function. @dukenv0307 is the idea that we'll eventually add to these functions to separate them?

**Comment by c3024** [comment]
RBR on `ReportPreview` disappears when we remove the receipt.

I think we gotta modify `ReportPreview` as well here.

https://github.com/Expensify/App/blob/cd148302d6b2266c9ac9ac36d9b7e2a337baa61b/src/components/ReportActionItem/ReportPreview.tsx#L141

https://github.com/Expensify/App/assets/102477862/1b6fc88c-dbbf-45ea-8607-3e47806b6502

@dukenv0307

**Comment by c3024** [comment]
@dukenv0307 In the test steps, please add a step between 4 and 5 as something like 
5. Verify that the error and RBR stay at the 'Merchant' field
and after 'Add a receipt' step as same
Verify that the error and RBR stay at the 'Merchant' field

**Review by c3024 — APPROVED** [review]
LGTM

**Review by Julesssss — APPROVED** [review]

**Pull request merged**